pkg/instrumentation/upgrade/upgrade.go

```diff
@@ -138,5 +139,13 @@ func (u *InstrumentationUpgrade) upgrade(_ context.Context, inst v1alpha1.Instru
 			u.Recorder.Event(inst.DeepCopy(), "Warning", "InstrumentationUpgradeRejected", "support for Go auto instrumentation is not enabled")
 		}
 	}
+	autoInstApacheHttpd := inst.Annotations[v1alpha1.AnnotationDefaultAutoInstrumentationApacheHttpd]
+	if autoInstApacheHttpd != "" {
+		// upgrade the image only if the image matches the annotation
+		if inst.Spec.ApacheHttpd.Image == autoInstApacheHttpd {
```

Comment:
We should add the feature-gate check here like the other languages

Reply:
Done.